actors/builtin/miner/miner_state.go

```diff
@@ -594,8 +602,10 @@ func (st *State) RemoveFaults(store adt.Store, sectorNos *abi.BitField) error {
 			return err
 		}
 
-		if countOld != countNew {
-			epochsChanged[uint64(i)] = epochFaultsNew
+		if countNew == 0 {
+			epochsDeleted = append(epochsDeleted, uint64(i))
+		} else if countOld != countNew {
+			epochsChanged = append(epochsChanged, change{index: uint64(i), value: epochFaultsNew})
```

Comment:
I assume that _deleting_ elements from an AMT while iterating isn't allowed, but is changing them allowed? That would allow us to update changed bit fields in-place.

Reply:
AFAIK it is ok, since the key isn't altered. But no docs :-(

But I think the pattern you have now is more clear, less tricky. If AMT one day supports a batch set, we'll want this anyway.